interfaces/cython/cantera/composite.py

```diff
@@ -904,9 +936,26 @@ def join(species):
             self._shape = (rows,)
 
         # restore data
-        for i in self._indices:
-            setattr(self._phase, mode, [st[i, ...] for st in state_data])
-            self._states[i] = self._phase.state
+        if normalize or mode[-1] == "Q":
+            for i in self._indices:
+                setattr(self._phase, mode, [st[i, ...] for st in state_data])
+                self._states[i] = self._phase.state
+        elif not normalize:
+            for i in self._indices:
+                if mode[-1] == "X":
+                    self._phase.set_unnormalized_mole_fractions([st[i, ...] for \
+                                                                 st in state_data][2])
```

Comment:
I think PEP 8 specifically dis-recommends using backslash line continuation. I think you can format this like:

```python
                    self._phase.set_unnormalized_mole_fractions(
                        [st[i, ...] for st in state_data][2]
                    )
```